packages/lib/src/tabs/Tabs.tsx

```diff
@@ -34,57 +52,70 @@ const DxcTabs = ({
+  const [countClick, setCountClick] = useState(0);
+  const [totalTabsWidth, setTotalTabsWidth] = useState(0);
   const [translateScroll, setTranslateScroll] = useState(0);
   const [scrollRightEnabled, setScrollRightEnabled] = useState(true);
   const [scrollLeftEnabled, setScrollLeftEnabled] = useState(false);
-  const [countClick, setCountClick] = useState(0);
-  const [totalTabsWidth, setTotalTabsWidth] = useState(0);
-  const [currentFocusIndex, setCurrentFocusIndex] = useState(activeTabIndex ?? innerActiveTabIndex);
-  const [temporalFocusIndex, setTemporalFocusIndex] = useState(activeTabIndex ?? innerActiveTabIndex);
   const [minHeightTabs, setMinHeightTabs] = useState(0);
-  const refTabs = useRef([]);
   const refTabList = useRef(null);
+  const colorsTheme = useTheme();
   const viewWidth = useResize(refTabList);
   const translatedLabels = useTranslatedLabels();
   const enabledIndicator = useMemo(() => viewWidth < totalTabsWidth, [viewWidth]);
 
+  const isActiveIndicatorDisabled = Children.toArray(children).some((child) => {
+    if (isValidElement(child)) {
+      return activeTabLabel === child.props.label && child.props.disabled;
+    }
+    return false;
+  });
+
   useEffect(() => {
-    let sumWidth = refTabs?.current?.reduce(function (count, obj) {
-      return count + obj.offsetWidth;
-    }, 0);
-    setTotalTabsWidth(sumWidth);
-    setActiveIndicatorWidth(refTabs?.current[activeTabIndex ?? innerActiveTabIndex]?.offsetWidth);
-    setActiveIndicatorLeft(refTabs?.current[activeTabIndex ?? innerActiveTabIndex]?.offsetLeft);
-  }, [refTabs]);
+    setTotalTabsWidth(refTabList?.current?.firstElementChild?.offsetWidth);
+  }, [refTabList]);
 
   useEffect(() => {
     setMinHeightTabs(refTabList?.current?.offsetHeight + 1);
   }, [refTabList]);
 
-  useEffect(() => {
-    if (activeTabIndex >= 0) {
-      setActiveIndicatorWidth(refTabs?.current[activeTabIndex]?.offsetWidth);
-      setActiveIndicatorLeft(refTabs?.current[activeTabIndex]?.offsetLeft);
-    }
-  }, [activeTabIndex]);
-
-  const handleSelected = (newValue) => {
-    activeTabIndex ?? setInnerActiveTabIndex(newValue);
-    onTabClick?.(newValue);
-    if (activeTabIndex === undefined) {
-      setActiveIndicatorWidth(refTabs?.current[newValue]?.offsetWidth);
-      setActiveIndicatorLeft(refTabs?.current[newValue]?.offsetLeft);
-    }
-  };
+  const contextValue = useMemo(() => {
+    const focusedChild = Children.toArray(children)[innerFocusIndex];
+    return {
+      iconPosition,
+      tabIndex,
+      focusedLabel: isValidElement(focusedChild) && focusedChild.props.label,
+      isControlled: childrenArray.some((child) => isValidElement(child) && typeof child.props.active !== "undefined"),
+      activeLabel: activeTabLabel,
+      hasLabelAndIcon: hasLabelAndIcon,
+      setActiveLabel: setActiveTabLabel,
+      setActiveIndicatorWidth,
+      setActiveIndicatorLeft,
+    };
+  }, [iconPosition, tabIndex, innerFocusIndex, activeTabLabel]);
```

Comment:
Theoretically, `hasLabelAndIcon` and `childrenArray` should go inside the dependency array, but, I think that may trigger an endless loop because their value comes from `Children`... I'm not sure about this implementation 🤔.

Reply:
I would encourage you to look over the usage of Children and try to avoid it as much as possible. I link you some alternatives:

https://react.dev/reference/react/Children#alternatives

Reply:
I think I found a workaround for the endless loop using memoizing. I have just added it, can you check it?